### DISCUSSION TITLE
How do we intend to represent CarePlans to manage Household tasks ?

### DISCUSSION THREAD

**Discussion by f-odhiambo** [opensrp/fhircore#534] [Ideas]
Based on the FHIR spec, a [CarePlan](https://www.hl7.org/fhir/careplan.html) are used in many areas of healthcare with a variety of scopes. They can be as simple as a general practitioner keeping track of when their patient is next due for a tetanus immunization through to a detailed plan for an oncology patient covering diet, chemotherapy, radiation, lab work and counselling with detailed timing relationships, pre-conditions and goals. 

They can represent a specific plan instance for a particular **patient** or **group**. _It is not intended to be used to define generic plans or protocols that are independent of a specific individual or group_. CarePlan represents a specific intent, not a general definition. Protocols and order sets are supported through [PlanDefinition](https://www.hl7.org/fhir/plandefinition.html).

This resource is referenced by itself, DiagnosticReport, GuidanceResponse, ImagingStudy, Media, MedicationRequest, MedicationStatement, Observation, Procedure, QuestionnaireResponse and ServiceRequest. 

Given that we are not representing a family as a group but reference to various patient resources, how do we intend to generate household tasks?

**Comment by f-odhiambo** [comment]
Do we assign it to the head of household and if so, what happens to his or her own individual CarePlan?

**Comment by maimoonak** [comment]
A careplan subject is Patient/Group. By definition we can associate it to the entity we are using to represent HH, which is Head in our case. A person can have multiple careplans which be categorized by 'category' field. An an individual also a patient can have different association with different type of programs/cadres, so keeping it separate, I feel is easier to manage and scale in future. 

However, one thing we may want to figureout, if a head of household dies or changes then what should happen to the future careplans?

And also how do we define the boundaries between [Careplan and EpisodeOfCare ](https://www.hl7.org/fhir/episodeofcare.html#bnr)

A head of household can be associated with two different plans
- House Cleanliness
- ANC
- PNC
- etc

They may keep added into or out of different services.

**Reply by pld** [reply]
Cool, yes this is exactly what we had discussed so far and I think this makes sense. The Household CarePlan is attached to the Head of Household and tagged as for a Household, any Tasks (and other resources) generated from the Household CarePlan are also tagged as for the Household so we can filter Household versus patient CarePlans, Tasks, etc

To handle death I see at least 2 options:
1) We also require a living Head of household, so if a Head of household dies, before they are marked as deceased, the CHW must reassign the Head of household to be someone else

are there any situations where the death of the House of household means there is a household which no longer has a head? We're assuming no and that implicit in the definition of a family is that there is always a single Head of household (this assumptions seems flawed in theory but maybe it's ok in practice)

2) When looking for Household tagged resources we have special business logic that ignores that a head of household is deceased

this seems like a bad idea because of all the special business logic it would require

**Reply by maimoonak** [reply]
The entities/attributes which hold the head of household info are
- Flag (subject, start/end/status represent the current status of Flag and remain there forever)
- Patient.tag/extension (when changing HHH we would need to remove the tag/extension from current Patient, and associate with new Patient, which would mean that person is no more a HHH and Flag still has info about all HHH in past/present).

We would also need to change the Patient.link into family member which actually associates the members to a HHH.

In any case we are loosing the individuality of HH. i.e. if a family member suffered death of his 2 heads consecutively, we can not track that.

January, 2020
Patient.A.tag=Head
Patient.1.link=Patient.A

February, 2020
Patient.B.tag=Head
Patient.1.link=Patient.B

March, 2020
Patient.C.tag=Head
Patient.1.link=Patient.C

How do we track that Patient.1 actually belongs to a Family(A,B,C)

**Reply by pld** [reply]
When patient A is unassigned as HH they are assigned as a family member w/HH patient B and so on.

Does that solve this?

**Reply by maimoonak** [reply]
@pld @f-odhiambo lets have a call on this. But yeah, we had missed that an unassigned HH should be added as a deceased member of family.

**Reply by pld** [reply]
Sounds good, yea when you're removing them as HH you'd want to remain in the family
